### PR TITLE
sql-insert functionality initial

### PR DESCRIPTION
**Pull request by shiraeeshi**

### CODE REVIEW

**Comment by mgolokhov** [comment]
Thank you for your contribution.
You have done a lot of work, so I need some time to check and test the code before merging.

**Comment by mgolokhov** [comment]
A very good code. I like your DB normalisation and "True or False" question optimisation.

Some questions popped in my mind.
When do we need to close Cursor and SQLiteDatabase?
Are their any advantages in using rawQuery over query/insert/..?

File: app/src/main/java/doit/study/droid/sqlite/helper/DatabaseHelper.java [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+        }
+        return questionInfos;
+    }
+
+    public int countQuestions() {
+        SQLiteDatabase db = this.getReadableDatabase();
+        String countQuery = "SELECT COUNT(*) FROM " + TABLE_QUESTION;
+        Cursor c = db.rawQuery(countQuery, null);
+        c.moveToFirst();
+        return c.getInt(0);
```
Comment (mgolokhov) [inline app/src/main/java/doit/study/droid/sqlite/helper/DatabaseHelper.java:210]: micro optimization:
DatabaseUtils.queryNumEntries (getReadableDatabase(), TABLE_QUESTION)

File: app/src/main/java/doit/study/droid/sqlite/helper/DatabaseHelper.java [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+        db.execSQL("DROP TABLE IF EXISTS " + TABLE_QUESTION_TAG);
+
+        // create new tables
+        onCreate(db);
+    }
+
+    public void importFromFile() {
+
+        File flagFile = new File(mContext.getFilesDir(), "imported");
+        if (flagFile.exists()) return;
```
Comment (mgolokhov) [inline app/src/main/java/doit/study/droid/sqlite/helper/DatabaseHelper.java:133]: To insert new questions need some counter like DATABASE_VERSION.
We can store it as Shared Preferences.

**Comment by shiraeeshi** [comment]
- "When do we need to close Cursor and SQLiteDatabase?"
  You're right, I've edited the code, committed to 'no-global' branch.
- "Are their any advantages in using rawQuery over query/insert/..?"
  I'm not sure if I understood the question right, I use template query with question marks in place of parameters in the case when query will be invoked in a loop. I guess it's might be more efficient to have such a template query, but it's just a guess.

**Comment by mgolokhov** [comment]
> Are their any advantages in using rawQuery over query/insert/..?" 

There are two strategies:
- create raw sql string and query it
- use SQLiteDatabase.query() / SQLiteDatabase.insert() with the help of ContentValues() / and so on.

The former can be (may be not) a little bit faster. The latter is more secure (prevent sql injections).

**Pull request merged**